app/models/services/locators/doi_paper_locator.rb

```diff
@@ -6,30 +6,79 @@ def column
   def find_paper
     return super if super
 
+    paper = Paper.new
+
+    crossref = Crossref.new
+    if (result = crossref.get_doi_metadata(self.locator_id))
```

Comment:
If it really actually catastrophically fails, then we should let it fail. Otherwise, we should handle 4xx and 5xx responses from the external API gracefully: https://github.com/briankung/projectcredo/pull/99/files/421f3270169e3dadbce1f99dff2edfbee31b426c#r82536967